src/main/java/parser/Function.java

```diff
@@ -223,7 +224,7 @@ public double calc(double... x) {
         return Double.NaN;
     }
 
-    public static boolean assignObject(String input, Class mathExpClass) {
```

Comment:
Mybe instead of extends MatheException, only Solvable interface may be expected?

Reply:
If solvable is not enough, then necessary interface methods can be added to it. Still better then expect Extending class.
But I was not reding whole class. Maybe Extending mathe expression is best here.

Reply:
> If solvable is not enough, then necessary interface methods can be added to it. Still better then expect Extending class. But I was not reding whole class. Maybe Extending mathe expression is best here.

These structural changes are too large to make in this PR.

Reply:
Maybe. Still it is worthy to try.

Reply:
I recommend that after merging this PR and then working on it further in another branch. (or in `add-bigdecimal-capability`)

Reply:
Well, yes. thats for sure.